src/main/scala/akka/persistence/cassandra/journal/CassandraJournal.scala

```diff
@@ -245,8 +250,38 @@ class CassandraJournal(cfg: Config) extends AsyncWriteJournal with CassandraReco
     session.executeAsync(stmt).map(_ => ())
   }
 
-  def partitionNr(sequenceNr: Long): Long =
-    (sequenceNr - 1L) / targetPartitionSize
+  private def readHighestSequenceNr(persistenceId: String, fromSequenceNr: Long): Long = {
+    findHighestSequenceNr(persistenceId, math.max(fromSequenceNr, highestDeletedSequenceNumber(persistenceId)))
+  }
+
+  private def readLowestSequenceNr(persistenceId: String, fromSequenceNr: Long): Long = {
+    new MessageIterator(persistenceId, fromSequenceNr, Long.MaxValue, Long.MaxValue).find(!_.deleted).map(_.sequenceNr).getOrElse(fromSequenceNr)
```

Comment:
I would prefer to address this separately with the FIXME below. This code has been around since April and I only moved it here from CassandraRecovery.